src/internal/snapshot/recovery.go

```diff
@@ -214,6 +215,65 @@ func (s *Snapshotter) dumpTx(ctx context.Context, tx *pachsql.Tx, opts CreateSna
 	return id, fh, nil
 }
 
+func (s *Snapshotter) pinDatabaseDump(ctx context.Context, fh io.Reader, id SnapshotID) (retErr error) {
+	ctx, done := log.SpanContext(ctx, "pinDatabaseDump")
+	defer done(log.Errorp(&retErr))
+
+	var closedFileSet, pinnedFileSet bool
+	fw := s.Storage.NewWriter(ctx)
+	defer func() {
+		if closedFileSet {
+			return
+		}
+		if _, err := fw.Close(); err != nil {
+			errors.JoinInto(&retErr, errors.Wrap(err, "close abandoned fileset"))
+		}
+	}()
+	if err := fw.Add(SQLDumpFilename, "", fh); err != nil {
+		return errors.Wrap(err, "create fileset containing database dump")
+	}
+	closedFileSet = true
+	fsHandle, err := fw.Close()
+	if err != nil {
+		return errors.Wrap(err, "close finished database dump fileset")
+	}
+	if fsHandle == nil {
```

Comment:
What is this for? Close is either non-nil handle with nil error, or nil handle with error.

Reply:
Removed.